tomviz/ModuleRuler.cxx

```diff
@@ -150,8 +150,10 @@ bool ModuleRuler::setVisibility(bool val)
 {
   vtkSMPropertyHelper(m_Representation, "Visibility").Set(val ? 1 : 0);
   m_Representation->UpdateVTKObjects();
-  if (m_Widget) {
+  if (!val || m_showLine) {
```

Comment:
I think this warrants a comment explaining what it is doing, I think I see that you are storing the member variable's value as it will be changed when you toggle the widget visibility. I don't understand why the null check for m_Widget was removed though.